### ISSUE TITLE
Use unsupervised learning to develop new groupers

### ISSUE THREAD

**Issue by jcrudy** [mattlewissf/mimic#6]
This is probably a sparse matrix problem of some kind.  There are some methods in scikit-learn that accept sparse data.  There are also methods that accept precomputed distance matrices.  I had some success using topic modeling with [gensim](https://radimrehurek.com/gensim/).  

To get started, you'll have to iterate over people and, for each one, get all of the icd-9 codes and whatever other relevant data.  You'll then transform these data into whatever form the method you're trying to use accepts, apply the method, and validate the results.  One way to validate the results is to fit a 30-day readmission model (or some other outcome: death, admission in the next year, heart attack, spending. There are many options) and see whether prediction improves (the [AUC](http://scikit-learn.org/stable/modules/generated/sklearn.metrics.roc_auc_score.html) is a good metric here).  There are also many other ways to assess the quality of a clustering or dimension reduction which you can look into.  I'm not expert on that topic.
